processor/metricextractprocessor/README.md

```diff
@@ -0,0 +1,93 @@
+
+### Example Config
+The following config is an example configuration of the metric extract processor using default values. In this example, logs are collected from a file, sent to the processor to have a `byte.count` metric extracted, and then consumed by the logging exporter. The metrics created are sent to the Google Cloud exporter.
+```yaml
+receivers:
+    filelog:
+        include: [./example/apache.log]
+    route/example:
+processors:
+    batch:
+    metricextract:
+        route: example
+        extract: body.byte_count
+        metric_name: byte.count
+        metric_unit: by
+        metric_type: gauge_int
+exporters:
+    googlecloud:
+    logging:
+
+service:
+    pipelines:
+        logs:
+            receivers: [filelog]
+            processors: [batch, metricextract]
+            exporters: [logging]
+        metrics:
+            receivers: [route/example]
+            processors: [batch]
+            exporters: [googlecloud]
+```
+
+## Expression Language
+In order to match or extract values from logs, the following `keys` are reserved and can be used to traverse the logs data model.
+
+| Key               | Description |
+| ---               | ---   |
+| `body`            | Used to access the body of the log. |
+| `attributes`      | Used to access the attributes of the log. |
+| `resource`        | Used to access the resource of the log. |
+| `severity_enum`   | Used to access the severity enum of the log. |
+| `severity_number` | Used to access the severity number of the log. |
+
+In order to access embedded values, use JSON dot notation. For example, `body.example.field` can be used to access a field two levels deep on the log body. 
+
+However, if a key already possesses a literal dot, users will need to use bracket notation to access that field. For example, when the field `service.name` exists on the log's resource, users will need to use `resource["service.name"]` to access this value.
+
+For more information about syntax and available operators, see the [Expression Language Definition](https://github.com/antonmedv/expr/blob/master/docs/Language-Definition.md).
+
+## How To
+### Match specific logs
+The following configuration adds a match expression that will count only logs with a 4xx status code. In this example, incoming logs are expected to have a `status` attribute.
+```yaml
+processors:
+    metricextract:
+        match: attributes.status startsWith "4"
+        extract: body.byte_count
+        metric_name: byte.count
+```
+
+### Extract metric attributes
+The following configuration extracts the status and endpoint values from the body of a log. These values are used as metric attributes. For each unique combination observed, a unique metric count is created.
```

Comment:
This is a mistake on my part. I started with the `logcount` processor README as a starting point, since the two processors share so much in common. This sentence was missed and should be removed.

Reply:
@jmwilliams89 so what is the intended behavior if there are multiple types of matches for an attribute across the logs?

Reply:
@jmwilliams89 i guess that makes sense and is fair...i still think it could be helpful as a reminder that there will be multiple metrics made as you had it, but i won't block on it and leave it up to you